### ISSUE TITLE
Rename submission file locale attribute to submissionLocale

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#9465]
The SubmissionFile object has 'locale' attribute/data. This is the submission's (primary) locale. It seems to be used correctly everywhere, but it would be better to call it 'submissionLocale'? -- One could think it is the locale/language of the file, which is not true -- the locale of the file is the galley locale -- i.e. it can be a different language than submission primary locale.

**Comment by asmecher** [comment]
@bozana, is it just a duplicate of the submission locale? If it doesn't have its own purpose, maybe it would be best to remove it.

**Comment by asmecher** [comment]
Just looking into this a little further -- the `SubmissionFile::getLocale/setLocale` are not persisted to the database. See [github](https://github.com/pkp/pkp-lib/blob/22793684471ca35bae39f059c51806c14ee6045e/classes/submissionFile/SubmissionFile.php#L66..L68):

> This is not properly a property of the submission file
> (e.g. it won't be persisted to the DB with the update function)
> It helps solve submission locale requirement for file's multilingual metadata

These attributes are in service of what has recently become the [`LocalizedData` trait](https://github.com/pkp/pkp-lib/blob/22793684471ca35bae39f059c51806c14ee6045e/classes/core/traits/LocalizedData.php#L23), which expects implementers to know what the `getDefaultLocale` is for the entity. In the case of `Author`s and `SubmissionFile`s, it should come from the submission; it's provided by the DAO (see the `fromRow` function in each case).

I've removed the `getSubmissionLocale` / `setSubmissionLocale` functions, which in the case of `SubmissionFile` were marked deprecated anyway. I've standardized on `submissionLocale` rather than `locale` in calls to `setData` / `getData` on the `DataObject` in order to make clear that these objects don't have their own independent locale column/attribute.
